In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/india-ev-market-data/Vehicle Class - All.csv
/kaggle/input/india-ev-market-data/ev_sales_by_makers_and_cat_15-24.csv
/kaggle/input/india-ev-market-data/EV Maker by Place.csv
/kaggle/input/india-ev-market-data/OperationalPC.csv
/kaggle/input/india-ev-market-data/ev_cat_01-24.csv


In [2]:
import pandas as pd
import sqlite3
place_path = "/kaggle/input/india-ev-market-data/EV Maker by Place.csv"
operationaIPC_path = "/kaggle/input/india-ev-market-data/OperationalPC.csv"
Vehicle_Class_path = "/kaggle/input/india-ev-market-data/Vehicle Class - All.csv"
ev_cat_path = "/kaggle/input/india-ev-market-data/ev_cat_01-24.csv"
ev_sales_by_makers_cat_path = "/kaggle/input/india-ev-market-data/ev_sales_by_makers_and_cat_15-24.csv"
conn = sqlite3.connect('/kaggle/working/Electric_Vehicle_Data.db')
EV_Maker_by_place = pd.read_csv(place_path)
OperationalIPC = pd.read_csv(operationaIPC_path)
Vehicle_Class_All = pd.read_csv(Vehicle_Class_path)
ev_cat_01_24 = pd.read_csv(ev_cat_path)
ev_sales_by_makers_and_cat15_24 = pd.read_csv(ev_sales_by_makers_cat_path)

**Adoption rate of electric vehicles over the years**

In [3]:
vecheli_sold = pd.DataFrame(ev_sales_by_makers_and_cat15_24)
column_sums = ev_sales_by_makers_and_cat15_24[['2015','2016','2017','2018','2019','2020','2021','2022','2023','2024']].sum()
column_sums_df = column_sums.reset_index()
column_sums_df.columns = ['Year', 'Total_EV_Sold']
column_sums_df['Growth_Rate'] = column_sums_df['Total_EV_Sold'].pct_change() * 100
column_sums_df['Growth_Rate'] = column_sums_df['Growth_Rate'].fillna(0)
column_sums_df.to_csv('ev_adoption.csv', index=False)
column_sums_df

Year  Total_EV_Sold  Growth_Rate
0  2015           7752     0.000000
1  2016          49249   535.307018
2  2017          87019    76.691913
3  2018         129763    49.120307
4  2019         165786    27.760610
5  2020         123770   -25.343515
6  2021         328854   165.697665
7  2022        1020533   210.330116
8  2023        1529234    49.846600
9  2024         978943   -35.984748

**Geographical Analysis**

In [4]:
state_counts = EV_Maker_by_place['State'].value_counts().reset_index()
state_counts.columns = ['State', 'State_Count']
state_counts.to_csv('geographical_Analysis.csv', index=False)
state_counts

State  State_Count
0      Maharashtra           15
1       Tamil Nadu           11
2        Karnataka            6
3          Haryana            6
4          Gujarat            5
5            Delhi            4
6        Rajasthan            4
7        Telangana            3
8      West Bengal            3
9      Uttarakhand            2
10  Andhra Pradesh            1
11  Madhya Pradesh            1
12          Punjab            1

**Manufacturer Performance**

In [5]:
manufacturer_performance = ev_sales_by_makers_and_cat15_24.groupby('Maker')[['2015', '2016', '2017', '2018', '2019', '2020', '2021', '2022', '2023', '2024']].sum()
manufacturer_performance['Manufacturer_Sales'] = manufacturer_performance.sum(axis=1)
manufacturer_performance['Manufacturer_Sales'] = manufacturer_performance['Manufacturer_Sales'].fillna(0)
Total_Sales = manufacturer_performance['Manufacturer_Sales'].sum()
manufacturer_performance['Market_Share (%)'] = (manufacturer_performance['Manufacturer_Sales'] / Total_Sales) * 100
manufacturer_performance = manufacturer_performance.reset_index()
manufacturer_performance = manufacturer_performance[['Maker','Manufacturer_Sales', 'Market_Share (%)']]
manufacturer_performance.to_csv('manufacturer_performance.csv', index=False)
print("Total sales across all manufacturers:", Total_Sales)


Total sales across all manufacturers: 4420903


**Trend of total EV manufacturing through 2015 to 2024 : 4420903**

**EV Sales by Category 15-24**

In [6]:
column_sums = ev_cat_01_24[['FOUR WHEELER (INVALID CARRIAGE)',
                                'HEAVY GOODS VEHICLE',
                                'HEAVY MOTOR VEHICLE',
                                'HEAVY PASSENGER VEHICLE',
                                'LIGHT GOODS VEHICLE',
                                'LIGHT MOTOR VEHICLE',
                                'LIGHT PASSENGER VEHICLE',
                                'MEDIUM GOODS VEHICLE',
                                'MEDIUM PASSENGER VEHICLE',
                                'MEDIUM MOTOR VEHICLE',
                                'OTHER THAN MENTIONED ABOVE',
                                'THREE WHEELER(NT)',
                                'TWO WHEELER (INVALID CARRIAGE)',
                                'THREE WHEELER(T)',
                                'TWO WHEELER(NT)',
                                'TWO WHEELER(T)']].sum()
column_sums_df = column_sums.reset_index()
column_sums_df.columns = ['Category', 'Total_EV_Manufactured']
column_sums_df.to_csv('EV_Manufactured by Category.csv', index=False)
column_sums_df
    

Category  Total_EV_Manufactured
0   FOUR WHEELER (INVALID CARRIAGE)                     97
1               HEAVY GOODS VEHICLE                    614
2               HEAVY MOTOR VEHICLE                    146
3           HEAVY PASSENGER VEHICLE                   8131
4               LIGHT GOODS VEHICLE                  12725
5               LIGHT MOTOR VEHICLE                 180330
6           LIGHT PASSENGER VEHICLE                  22807
7              MEDIUM GOODS VEHICLE                     31
8          MEDIUM PASSENGER VEHICLE                    776
9              MEDIUM MOTOR VEHICLE                     49
10       OTHER THAN MENTIONED ABOVE                   7589
11                THREE WHEELER(NT)                   1498
12   TWO WHEELER (INVALID CARRIAGE)                    126
13                 THREE WHEELER(T)                1932683
14                  TWO WHEELER(NT)                2308887
15                   TWO WHEELER(T)                  13459